docs/javascript/v2/advanced-features/volume-control.mdx

```diff
@@ -16,4 +22,76 @@ hmsActions.setVolume(volume);
 
 // set the volume for a specific track
 hmsActions.setVolume(volume, "<track_id>");
+
+// get the currrent volume for a specific track
+hmsStore.getState(selectAudioTrackVolume("<track_id>"));
+```
+
+</Tab>
+
+<Tab id='volume-1'>
+
+```jsx
+function PeerVolumeController({peerID}) {
+  const track = useHMSStore(selectAudioTrackByPeerID(peerID));
+  const currVolume = useHMSStore(selectAudioTrackVolume(track?.id));
+  const hmsActions = useHMSActions();
+
+  const setVolume = (volume) => {
+    if (track) {
+      console.log(`setting volume for track=${track.id}, prev=${currVolume}, new=${volume}`);
+      hmsActions.setVolume(volume, track.id);
+    }
+  }
+
+  return <Slider onChange={setVolume}/>;
+}
+```
+
+</Tab>
+
+## Gotchas
+
+Availability of peer doesn't mean availability of their audio track. The below code can silence the whole room if peer's audio track is not yet present.
+
+```js
+const peer = hmsStore.getState(selectPeerById("peer-id"));
+hmsActions.setVolume(0, peer?.audioTrack);  // 
+```
+
+If you want to set volume for specific track, always do a check before calling setVolume - 
+
+
+```js
+const peer = hmsStore.getState(selectPeerById("peer-id"));
+if (peer?.audioTrack) {
+  hmsActions.setVolume(0, peer.audioTrack);
+}
+```
+
+## Ideas
```

Comment:
love this section <3

Reply:
it's part of a few other pages as well - https://www.100ms.live/docs/javascript/v2/features/change-role#ideas
https://www.100ms.live/docs/javascript/v2/advanced-features/peer-metadata#ideas